coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -451,7 +453,22 @@ protected SqlNode visitNumber(ASTNode node, ParseContext ctx) {
 
   @Override
   protected SqlNode visitAllColRef(ASTNode node, ParseContext ctx) {
-    return SqlIdentifier.star(ZERO);
+    List<SqlNode> children = visitChildren(node, ctx);
+    // This is to allow t.* and t.col.*
+    // In Hive ASTNode Tree, "t.*" has the following shape
+    // TOK_ALLCOLREF
+    // - TOK_TABLE_OR_COL
+    //   - "t"
+    List<String> names = new ArrayList<>();
+    if (children != null) {
+      for (SqlNode child : children) {
+        names.addAll(((SqlIdentifier) child).names);
+      }
+    }
+    names.add("*");
+    List<SqlParserPos> sqlParserPos = Collections.nCopies(names.size(), ZERO);
+    SqlNode star = SqlIdentifier.star(names, ZERO, sqlParserPos);
+    return star;
```

Comment:
nit: you can inline `star` variable

Reply:
Yeah this doesn't hurt though.  It makes debugging a bit easier: set breakpoint and examine the value of `star` in IDE.  I hope this is OK.